hibernate-core/src/main/java/org/hibernate/cfg/ToOneFkSecondPass.java

```diff
@@ -66,19 +66,32 @@ else if ( property != null) {
 			//try explicit identifier property
 			return path.startsWith( property.getName() + "." );
 		}
-		else {
-			//try the embedded property
-			//embedded property starts their path with 'id.' See PropertyPreloadedData( ) use when idClass != null in AnnotationSourceProcessor
-			if ( path.startsWith( "id." ) ) {
-				KeyValue valueIdentifier = persistentClass.getIdentifier();
-				String localPath = path.substring( 3 );
-				if ( valueIdentifier instanceof Component ) {
-					Iterator it = ( (Component) valueIdentifier ).getPropertyIterator();
-					while ( it.hasNext() ) {
-						Property idProperty = (Property) it.next();
-						if ( localPath.startsWith( idProperty.getName() ) ) return true;
+		//try the embedded property
+		//embedded property starts their path with 'id.' See PropertyPreloadedData( ) use when idClass != null in AnnotationSourceProcessor
+		else if ( path.startsWith( "id." ) ) {
+			KeyValue valueIdentifier = persistentClass.getIdentifier();
+			String localPath = path.substring( 3 );
+			if ( valueIdentifier instanceof Component ) {
+				Iterator it = ( (Component) valueIdentifier ).getPropertyIterator();
+				while ( it.hasNext() ) {
+					Property idProperty = (Property) it.next();
+					if ( localPath.startsWith( idProperty.getName() ) ) {
+						return true;
+					}
+				}
+			}
+		}
+		// Try the case where a @ManyToOne is also an ID property
+		// E.g. @ManyToOne @Id SomeEntity other;
+		else if ( !path.contains( "." ) ) {
```

Comment:
In the debugger, I'm seeing @OneToOne @Id from `EntityWithOneToOneDerivedId#idsource` falling into this area.

Reply:
The comment is wrong, I also meant to cover `@OneToOne` here.